aiohttp/web_urldispatcher.py

```diff
@@ -637,27 +669,26 @@ class PrefixedSubAppResource(PrefixResource):
     def __init__(self, prefix, app):
         super().__init__(prefix)
         self._app = app
-        for resource in app.router.resources():
-            resource.add_prefix(prefix)
-
-    def add_prefix(self, prefix):
-        super().add_prefix(prefix)
-        for resource in self._app.router.resources():
-            resource.add_prefix(prefix)
+        for subresource in app.router.resources():
+            subresource.add_prefix(self)
 
-    def url_for(self, *args, **kwargs):
-        raise RuntimeError(".url_for() is not supported "
-                           "by sub-application root")
+    def add_prefix(self, resource):
+        super().add_prefix(resource)
+        for subresource in self._app.router.resources():
+            subresource.add_prefix(resource)
 
     def get_info(self):
         return {'app': self._app,
                 'prefix': self._prefix}
 
     async def resolve(self, request):
-        if not request.url.raw_path.startswith(self._prefix):
+        if not request.url.path.startswith(self._prefix):
```

Comment:
Why not `raw_path`?

Reply:
What if the prefix includes arbitrary unicode characters?

Reply:
Mixing PCT encoded and decoded data can make a mess easy.
I believe we should follow the strategy:
1. Convert all user provided *path*, *prefix* etc into *normalized* (PCT-encoded actually) form by using `yarl`. Already normalized paths passed *as-is* by yarl.
2. Normalize given HTTP path (we do it).
3. Match normalized path to already pre-normalized patterns.
4. Return `request.match_info['name']` in **PCT-decoded** form because users most likely want getting `이름` instead of `'%EC%9D%B4%EB%A6%84`

If aiohttp code violates the rule somewhere -- it should be fixed.
Do you agree?

Reply:
Okay, I agree to keep the rule consistent. Let me inspect and fix up this PR.

Reply:
I think then `PlainResource.__init__()` also need this, like `PrefixResource`:
```python
self._path = URL.build(path=path).raw_path
```